Lara-JS/src-api/weaver/JoinPoints.ts

```diff
@@ -0,0 +1,214 @@
+import { LaraJoinPoint, wrapJoinPoint } from "../LaraJoinPoint.js";
+import Weaver from "./Weaver.js";
+
+/**
+ * Object which provides low-level join point-related methods.
+ */
+export default class JoinPoints {
+  /**
+   *
+   * @returns the current root node of the AST
+   */
+  static root(): LaraJoinPoint {
+    return wrapJoinPoint(Weaver.getWeaverEngine().getRootJp());
+  }
+
+  /**
+   * Converts an AST node to a JointPoint.
+   *
+   */
+  static toJoinPoint(node: any): LaraJoinPoint {
+    throw "JoinPoints.toJoinPoint: not implemented";
+  }
```

Comment:
This was done to maintain compatibility with current `Joinpoints` implementations on the weavers. As they were all very similar or the same, the `JoinpointsBase` class has been made final and now includes all the methods that the weavers added that were common between them. This can be removed though.